src/query_graph/extract_subgraphs_from_supergraph.rs

```diff
@@ -2036,3 +2036,35 @@ fn is_external_or_has_external_implementations(
     }
     Ok(false)
 }
+
+static DEBUG_SUBGRAPHS_ENV_VARIABLE_NAME: &str = "APOLLO_FEDERATION_DEBUG_SUBGRAPHS";
+
+fn maybe_dump_subgraph_schema(subgraph: FederationSubgraph, message: &mut String) {
+    // NOTE: The std::fmt::write returns an error, but writing to a string will never return an
+    // error, so it is dropped when called.
+
+    if let Some(true) = std::env::var(DEBUG_SUBGRAPHS_ENV_VARIABLE_NAME)
```

Comment:
I think this warning message should be printed for false/undefined case? In JS we do `!!<envBoolValue>` which also correctly handles `undefined`

Reply:
Oops, you're right. Addressed it in 4e7d7c9b59875f3a1f97feb657de858ba28d9f78